src/components/Profile.js

```diff
@@ -30,26 +30,82 @@ class Profile extends Component {
       email: decoded.identity.email,
     });
   }
+ 
+// deleteRow(id) {
 
-  deleteRow(name) {
-    const index = this.state.uesr_items.findIndex((user_items) => {
-      return user_items.name === name;
-    });
-    console.log("index", index);
+//     this.setState({
+
+//       posts: [...this.state.posts.filter(post => post.id !== id)]
+
+//     });
+
+//   }
+  deleteRow(name, email) {
+    // console.log(name_match);
+    // // this.setState({items:name_match})
+    // this.setState((items) =>({items: name_match}));
+    // console.log(name_match);
+    // console.log('name', name)
+    
+//     axios
+//       .delete("http://localhost:5000/users/" + encodeURIComponent(email.email))
+//       .then((response) => {
+    //     const index = this.state.items.find((user_items) => {
+    //       return user_items.name === name;
+
+    // })
+    // this.setState({items: name},() => console.log(this.state.items))
+    // this.setState({items: name});
+    // var array = this.state.items
+    // console.log("ARRAY", array);
+    // console.log("NAME", name);
+    // console.log("DELETE ROW ARRAY", array.user_items);
+    // console.log(email);
+    axios
+    // .delete("/users/" + "ken@gmail.com" + "/items",
+    .delete("/users/" + encodeURIComponent(email) + "/items",
+    {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      data: name
+    })
+    .then((response) => {
+      const token = window.localStorage.getItem("usertoken");
+      const decoded = jwt_decode(token);
+      this.getItem({
+        email: decoded.identity.email,
+      });
+    })
+    .catch((error) => {
+      console.log(error);
+      alert("Could not delete data");
+    }); 
+
+  // componentDidUpdate(prevProps, prevState){
+  //   console.log("THIS STATE", this.state.items.constructor !== Array)
+  //   console.log("THIS STATE", this.state.items)
+  //   console.log("prev state", prevState.items)
+  //   // if (this.state.items.constructor !== Array && this.state.items === prevState.items){
+  //   //   console.log("State matches Item");
+    // }
   }
+
   getItem(email) {
     axios
-      .get("http://localhost:5000/users/" + encodeURIComponent(email.email))
+      // .get("http://localhost:5000/users/" + encodeURIComponent(email.email))
+      .get("/users/" + encodeURIComponent(email.email))
       .then((response) => {
         const user_items = response.data;
         console.log(user_items);
         JSON.stringify(user_items);
         console.log(JSON.stringify(user_items));
         this.setState(user_items);
-        console.log("Data received");
+        this.setState({items: user_items})
```

Comment:
`setState()` is called one line above, this can be merged into the previous line.

Reply:
Turns out that suggestion breaks the code. It has something to do with an ReactTable internal function being unable to map our email parameter (currently obj, needs array). I tried changing the email value to an array but it still wouldn't work.

The question was, how come `this.setState(user_items);` still works? **If you don't set it to a value in the constructor, it sets it for you.** Note items vs user_items
![image](https://user-images.githubusercontent.com/13722143/90221147-a1c57180-ddd7-11ea-9cc9-83555c1e4482.png)

So instead of trying to debug and change it to that, I figure to just leave it since it works for now but quickly change to react-table 7 (or vanilla React/library)

Learned something new @kevmo314